### PR TITLE
Fund dapps

### PR DESCRIPTION
**Pull request by jstarry**
Fixes https://github.com/solana-labs/solana/issues/4858

- Checks if the requesting wallet is in the same network
- Adds validation for sufficient funds
- CSS changes to allow for scrollable mini window

---
Example dApp "Add Funds" CTA
Screenshot 2019-07-11 10 03 23
Mini wallet window
Screenshot 2019-07-11 10 03 55
Sufficient funds validation
Screenshot 2019-07-11 10 04 13
Network url must match
Screenshot 2019-07-11 10 04 56

### CODE REVIEW

File: src/wallet.js [hunk 1]
```diff
@@ -304,6 +321,16 @@ export class Wallet extends React.Component {
 
   componentDidMount() {
     this.props.store.onChange(this.onStoreChange);
+    const params = (new URL(document.location)).searchParams;
```
Comment (mvines) [inline src/wallet.js]: Can you please create a section in the README that documents the expected search params.

**Comment by jstarry** [comment]
> It looks like there's no way for the wallet to signal back to the dapp that the funding was completed, or rejected?

@mvines Good point, I'll change my approach to use window messages so that they can communicate with each other.

File: src/wallet.js (hunk 1 above)
Comment (jstarry) [reply src/wallet.js]: I'll change this to be exposed via window messages and will document the accepted message format

**Comment by jstarry** [comment]
@mvines can you take another look? The web wallet now notifies the dApp when the transaction succeeded or failed. Also, instead of checking url params, the wallet listens for post messages from the dApp.

**Comment by mvines** [comment]
Nice, I'll take a look tomorrow morning :)

File: README.md [hunk 2]
```diff
@@ -29,3 +29,57 @@ will rebuild and reload the app automatically
...
+    }
+  }
+});
+```
+4. Send an `'addFunds'` request
+```js
+walletWindow.postMessage({
+  method: 'addFunds',
+  params: {
+    pubkey: /* ... */,
```
Comment (mvines) [inline README.md]: nit: For these three can you give a concrete example, maybe in the comment, of what an accepted value would look like for a bit more context.

**Pull request merged**